src/main/java/com/example/WeatherDataCodeChallenge/SensorService.java

```diff
@@ -15,22 +15,17 @@
         return sensorRepository.findAll();
     }
 
-    //Dub key error
     public Sensor addNewSensor(Sensor sensor){
-        return sensorRepository.insert(sensor);
+        return sensorRepository.save(sensor);
     }
 
     public Sensor findById(String id) {
         Sensor sensor = getAllSensors().stream().filter(t -> id.equals(t.getSensorId())).findFirst().orElse(null);
         return sensor;
     }
```

Comment:
MongoRepository has a built-in `findAllById()` that you'd be able to use instead of this if you put the `@Id` annotation back on the sensorId. It'll mean the `save()` call in `addNewSensor` actually updates the existing entry rather than create a new sensor with the same ID (which I've noticed while testing it's doing currently)